src/audio/tdfb/tdfb_direction.c

```diff
@@ -282,9 +281,10 @@ static void level_update(struct tdfb_comp_data *cd, int frames, int ch_count, in
 	/* Calculate mean square level */
 	for (n = 0; n < frames; n++) {
 		s = *p;
+		tmp += ((int32_t)s * s);
```

Comment:
Good  catch, without the Q1.15 multiply shifts this indeed fits the int32_t type. Please put this also to to own commit.